### PR TITLE
chore: Update version to 24.0

### PR DESCRIPTION
**Pull request by mshabarov**

### CODE REVIEW

File: flow-plugins/flow-gradle-plugin/build.gradle [hunk 1]
```diff
@@ -132,7 +132,8 @@ task functionalTest(type: Test) {
         showStandardStreams = true
     }
 }
-check.dependsOn functionalTest
```
Comment (Artur-) [inline flow-plugins/flow-gradle-plugin/build.gradle:135]: Why do tests need to be disabled when changing the version number?

File: pom.xml [hunk 2]
```diff
@@ -67,14 +67,14 @@
             https://maven.vaadin.com/vaadin-addons
         </flow.addons.repo.url>
 
-        <maven.compiler.release>11</maven.compiler.release>
```
Comment (Artur-) [inline pom.xml:70]: Can't this be a separate PR for more clarity

**Review by Artur- — CHANGES_REQUESTED** [review]

File: flow-plugins/flow-gradle-plugin/build.gradle (hunk 1 above)
Comment (mshabarov) [reply flow-plugins/flow-gradle-plugin/build.gradle:135]: Because PR validation won't pass otherwise.

File: pom.xml (hunk 2 above)
Comment (mshabarov) [reply pom.xml:70]: ok, I'll make a separate PR.

File: flow-plugins/flow-gradle-plugin/build.gradle (hunk 1 above)
Comment (mshabarov) [reply flow-plugins/flow-gradle-plugin/build.gradle:135]: Making Gradle plugin support JDK 17 is a separate work, that we start soon.

**Review by Artur- — APPROVED** [review]
